biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java

```diff
@@ -2981,4 +2981,19 @@ public String getChecksum() {
 			throw Exceptions.duck(e);
 		}
 	}
+
+	/**
+	 * A macro to cleanup a version string (like Maven versions) to an OSGi
+	 * version
+	 */
+	public String _version_cleanup(String args[]) {
```

Comment:
This should use the more modern style:

```java
static final String _version_cleanupHelp = "${version_cleanup;<version>}, Cleanup a potential maven version to make it match the OSGi Version syntax.";

public String _version_cleanup(String args[]) throws Exception {
  Macro.verifyCommand(args, _version_cleanupHelp, null, 2, 2);
  return Analyzer.cleanupVersion(args[1]);
}
```

You should also push this macro down even further into the `Macro` class since this macro actually needs nothing from `Processor`.